language/stdlib/modules/Libra.move

```diff
@@ -1153,38 +1142,71 @@ module Libra {
         apply Roles::AbortsIfNotTreasuryCompliance{account: tc_account} to register_SCS_currency<CoinType>;
 
-        /// The permission "PreburnCurrency(type)" is granted to DesignatedDealer [B14].
-        apply AbortsIfNotDesignatedDealer to publish_preburn_to_account<CoinType>;
+        /// The permission "PreburnCurrency(type)" is granted to DesignatedDealer [B13].
+        apply Roles::AbortsIfNotDesignatedDealer to publish_preburn_to_account<CoinType>;
 
-        /// The permission "UpdateExchangeRate(type)" is granted to TreasuryCompliance [B15].
+        /// The permission "UpdateExchangeRate(type)" is granted to TreasuryCompliance [B14].
         apply Roles::AbortsIfNotTreasuryCompliance{account: tc_account} to update_lbr_exchange_rate<FromCoinType>;
     }
 
-    spec schema TotalValueRemainsSame<CoinType> {
-        /// The total amount of currency stays constant. The antecedant excludes register_currency
-        /// and register_SCS_currency, because they make start the currency with total_value = 0.
+    spec schema TotalValueNotIncrease<CoinType> {
+        /// The total amount of currency does not increase.
         ensures old(spec_is_currency<CoinType>())
-            ==> spec_currency_info<CoinType>().total_value == old(spec_currency_info<CoinType>().total_value);
+            ==> spec_currency_info<CoinType>().total_value <= old(spec_currency_info<CoinType>().total_value);
     }
+    spec schema TotalValueNotDecrease<CoinType> {
+        /// The total amount of currency does not decrease.
+        ensures old(spec_is_currency<CoinType>())
+            ==> spec_currency_info<CoinType>().total_value >= old(spec_currency_info<CoinType>().total_value);
+    }
+    spec schema PreburnValueNotIncrease<CoinType> {
+        /// The preburn value of currency does not increase.
+        ensures old(spec_is_currency<CoinType>())
+            ==> spec_currency_info<CoinType>().preburn_value <= old(spec_currency_info<CoinType>().preburn_value);
+    }
+    spec schema PreburnValueNotDecrease<CoinType> {
+        /// The the preburn value of currency does not decrease.
+        ensures old(spec_is_currency<CoinType>())
+            ==> spec_currency_info<CoinType>().preburn_value >= old(spec_currency_info<CoinType>().preburn_value);
+    }
+    spec schema ExchangeRateRemainsSame<CoinType> {
+        /// The exchange rate to LBR stays constant.
+        ensures old(spec_is_currency<CoinType>())
+            ==> spec_currency_info<CoinType>().to_lbr_exchange_rate == old(spec_currency_info<CoinType>().to_lbr_exchange_rate);
+    }
+
     spec module {
-        /// Only mint and burn functions can change the total amount of currency.
-        apply TotalValueRemainsSame<CoinType> to *<CoinType>
-            except mint<CoinType>, mint_with_capability<CoinType>,
-            burn<CoinType>, burn_with_capability<CoinType>, burn_with_resource_cap<CoinType>,
+        /// Only mint functions can increase the total amount of currency [B11].
```

Comment:
I feel like we're less likely to miss something if you put the apply in its own spec module right after each "...NotDecrease", or whatever, predicate.  Otherwise, a reviewer has to scroll back and forth to check that all predicates are appropriately applied.

Reply:
Rearranged them to put the "apply" statement right after the "schema" definition.